recipes/server.rb

```diff
@@ -63,9 +63,31 @@
                    SqlServer::Helper.sql_server_checksum(version, x86_64) ||
                    Chef::Application.fatal!("No package checksum matches '#{version}'. node['sql_server']['server']['checksum'] must be set or node['sql_server']['version'] must match a supported version.")
 
-windows_package package_name do
+filename = File.basename(package_url).downcase
+fileextension = File.extname(filename)
+is_iso = ['.iso'].include? fileextension
+
+if is_iso
+  include_recipe '7-zip' 
+end
+
+download_path = "#{Chef::Config['file_cache_path']}/#{filename}"
+remote_file download_path do
   source package_url
   checksum package_checksum
+  only_if {is_iso}
+end
+
+iso_extraction_dir = "#{Chef::Config['file_cache_path']}/#{package_checksum}"
```

Comment:
You should change package_checksum to a shorter file name.  when using a large checksum 7-Zip fails to copy files from the ISO file.  This is due the the Windows character limit.  I just experienced and this error and had to edit your package.  

I would also add a check to see if the files have and directory have already been created so Chef doesn't need to copy the files everything the recipe runs.  

Not job this really helped me with my sql_package.